ament_cpplint/ament_cpplint/main.py

```diff
@@ -144,7 +148,14 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += files
+        if args.excludedirs is not None:
+            filter_names = []
+            for f in files:
+                if any(exclude_dir not in f for exclude_dir in args.excludedirs):
+                    filter_names.append(f)
+            arguments += filter_names
+        else:
+            arguments += files
```

Comment:
Please simplify the logic to a single `arguments += ...` with a list comprehension.